src/containers/SearchContainer/index.js

```diff
@@ -0,0 +1,213 @@
+//             ...
+//           },
+//         },
+//       },
+//     },
+//     ...
+//   ]
+const getPositions = (searchResult, field) => {
+  let positions = [];
+  if (searchResult.matchData && searchResult.matchData.metadata) {
+    const data = searchResult.matchData.metadata;
+    Object.keys(data).forEach(searchTerm => {
+      if (data[searchTerm][field] && data[searchTerm][field].position) {
+        positions = positions.concat(data[searchTerm][field].position);
+      }
+    });
+  }
+  return positions;
+};
+
+const getSearchResults = query => {
+  let searchResults = [];
+  const categoryMap = { 'All Results': true };
+
+  if (query && window.__LUNR__ && window.__LUNR__.en) {
+    try {
+      const queryResults = window.__LUNR__.en.index.search(query);
+
+      searchResults = queryResults.map(searchResult => {
+        const doc = window.__LUNR__.en.store[searchResult.ref];
+        categoryMap[doc.sourceInstanceName] = true;
+        return {
+          title: getPositions(searchResult, 'title'),
+          body: getPositions(searchResult, 'body'),
+          tags: getPositions(searchResult, 'tags'),
+          doc,
+        };
+      });
+    } catch (x) {
+      // bad query e.g. title:
+    }
+  }
+
+  return { searchResults, searchCategories: Object.keys(categoryMap).sort() };
+};
+
+const SearchContainer = ({ location }) => {
+  const { term } = useParams(location);
+  const initialSearch = getSearchResults(term);
+  const [value, setValue] = useState(term);
+  const [results, setResults] = useState(initialSearch.searchResults);
+  const [categories, setCategories] = useState(initialSearch.searchCategories);
+  const [activeCategoryIndex, setActiveCategoryIndex] = React.useState(0);
+
+  const onChange = event => {
+    const { value: newValue } = event.target;
+    setValue(newValue);
+
+    // update the URL
+    const query = newValue ? `?term=${encodeURIComponent(newValue)}` : '';
+    navigate(`/search${query}`, { replace: true });
+
+    const { searchResults, searchCategories } = getSearchResults(newValue);
+    setResults(searchResults);
+    setCategories(searchCategories);
+    // todo update route term= param
+  };
+
+  const filterResults = category =>
+    results.filter(
+      ({ doc }) =>
+        !category ||
+        category === 'All Results' ||
+        doc.sourceInstanceName === category,
+    );
+
+  const onCategoryChange = index => setActiveCategoryIndex(index);
+
+  return (
+    <Box flex overflow="auto" gap="medium" pad="small">
+      <Box flex={false} direction="row-responsive" wrap>
+        <Box pad={{ vertical: 'large', horizontal: 'large' }}>
+          <Heading margin="none">Search</Heading>
+        </Box>
+        <Box pad={{ vertical: 'large' }} gap="medium">
+          <Box
+            width="large"
+            direction="row"
+            align="center"
+            pad={{ horizontal: 'small', vertical: 'xsmall' }}
+            border={{
+              side: 'all',
+            }}

```

Comment:
This should work the same with simply writing `border`, it will be assigned as true and all sides will be applied.
I just checked the docs and they are missing the boolean value for Box. I've filed an issue for it https://github.com/grommet/grommet/issues/3611 feel free to make a contribution :)

Reply:
Fun. Thanks :)